pkg/apis/extensions/v1beta1/defaults.go

```diff
@@ -87,6 +87,10 @@ func addDefaultingFuncs(scheme *runtime.Scheme) {
 				obj.Spec.UniqueLabelKey = new(string)
 				*obj.Spec.UniqueLabelKey = DefaultDeploymentUniqueLabelKey
 			}
+			if obj.Spec.MaxProgressThresholdSeconds == nil {
+				obj.Spec.MaxProgressThresholdSeconds = new(int32)
+				*obj.Spec.MaxProgressThresholdSeconds = 60
```

Comment:
My thought was that the default should be `nil`. Today's default is effectively "try forever" which I think is reasonable.